### ISSUE TITLE
Checks trigger for missing translations with activated translation propagation with multiple components

### ISSUE THREAD

**Issue by BhaaLseN** [WeblateOrg/weblate#1911]
### Steps to reproduce
1. Prepare a VCS repository in a way that translations are seperated in two (or more) folders and share the same resource keys (so translation propagation can update them), but make sure that one of the folders does not have all translations consistent (ie.: both folder1 and folder2 have a resource key "name", where folder1 is fully translated but folder2 has the English value in its French translation file)
2. Configure a Project with 2 Components; one for the first folder (using the real repository path) and one for the second folder (using the Weblate-internal URL to share the repository; not sure if that has any effect) while enabling the option "Allow translation propagation"
3. Navigate to Component1 > French and check the result of the "Source and translation are identical" check.

### Actual behaviour
The resource key "name" shows up as untranslated; but selecting it shows it correctly as translated. It does have the check next to it though.
It seems that right now, the check shows a mix of two things:
1. Translations that are actually untranslated within the currently selected component (which is the ones we want!)
2. Translations that are untranslated in any of the other components (which seems incorrect when viewing a specific component)

### Expected behaviour
We'd expect that only the current component is considered, regardless of repository sharing (or translation propagation).
Picking Component1 > French should show untranslated ones from Component1, while Component2 > French should show untranslated ones from Component2 (and not across the whole project for everything).

### Server configuration
```
$ ./manage.py list_versions
 * Weblate weblate-2.19.1
 * Python 3.5.3
 * Django 2.0.3
 * six 1.11.0
 * social-auth-core 1.7.0
 * social-auth-app-django 2.1.0
 * django-appconf 1.0.2
 * Translate Toolkit 2.2.5
 * Whoosh 2.7.4
 * defusedxml 0.5.0
 * Git 2.11.0
 * Pillow (PIL) 1.1.7
 * dateutil 2.7.0
 * lxml 4.2.0
 * django-crispy-forms 1.7.2
 * compressor 2.2
 * djangorestframework 3.7.7
 * user-agents 1.1.0
 * pytz 2018.3
 * pyuca N/A
 * python-bidi 0.4.0
 * PyYAML 3.12
 * Database backends: django.db.backends.mysql
```

### Additional Information
When picking Component1 > French and selecting the entry for "name", the translation is correct; but the check is still shown.
The "Other translations" tab correctly shows all components and their values, with Component2 being listed as different.

**Comment by nijel** [comment]
It's caused by way checks are stored, but indeed these checks should not be shown for non translated strings

**Comment by nijel** [comment]
Should be addressed once fixing #3007
